src/modules/download.ts

```diff
@@ -191,17 +194,22 @@ export class DownloadManager extends EventEmitter {
           download.progress.downloaded = transferred
         })
 
-        const res = await request
-
         try {
           await fs.mkdir(path.dirname(absolutePath), { recursive: true })
-          await fs.writeFile(absolutePath, res.rawBody)
+          await stream.pipeline(request, createWriteStream(absolutePath), {
+            signal: reqAc.signal,
+          })
```

Comment:
I've opted to just propagate them as they are, as the outer catch already handles them correctly. 

I've also propageted the AbortError so that it can be handled toghether with CancelError, 'cause they should be handled the same way.